aspnetcore/test/dev-tunnels.md

```diff
@@ -0,0 +1,169 @@
+
+Choose not to use a tunnel by selecting **None** in the flyout. When Visual Studio is restarted, it defaults back to **None**.
+
+## Use a tunnel
+
+When a tunnel is active and Visual Studio runs a web app, the web browser opens to a tunnel URL instead of a localhost URL. The tunnel URL looks like the following example:
+
+`https://0pbvlk3m-7032.usw2.devtunnels.ms`
+
+Now any authenticated user can open the same URL on any other internet-connected device. As long as the project continues to run locally, any device with an internet connection can access the web application that is running on a development machine.
+
+For web projects that have browser support, a warning page is shown on the first request sent to the tunnel URL from each device:
+
+:::image type="content" source="~/test/dev-tunnels/_static/dev-tunnels-notification.png" alt-text="Dev Tunnels notification page.":::
+
+After **Continue** is selected, the request is routed to the local web app. This notification page isn't shown for API requests using dev tunnels.
+
+### Use a tunnel to test on a phone or tablet
+
+To test a web app on an external device like a phone or tablet, navigate to the tunnel URL on the device. To make it easier to reproduce the URL on the external device:
+
+* Make sure that the Edge browser is on the tunnel URL.
+* Generate a QR code to the URL in the Edge browser:
+  * Select the URL bar and the QR code button appears.
+  * Select the QR code button to generate and view the QR code.
+    :::image type="content" source="~/test/dev-tunnels/_static/generate-qr-code.png" alt-text="QR code with button to create it highlighted.":::
+* Scan this QR code with a phone or tablet to navigate to the URL.
+
+## Dev Tunnels output window
+
+The **Dev Tunnels** output window shows the port and URL of a tunnel when a project starts using it. Select the **Dev Tunnels** output window by using the **Show output from:** dropdown in the **Output** window.
+
+:::image type="content" source="~/test/dev-tunnels/_static/dev-tunnels-output-window.png" alt-text="Dev Tunnels output window.":::
+
+This window is especially useful for projects that don't open a browser by default. For example, when working with an Azure Function, this may be the easiest way to discover the public URL that is being used by the dev tunnel.
+
+## Dev Tunnels tool window
+
+View and manage dev tunnels in the **Dev Tunnels** tool window:
+
+:::image type="content" source="~/test/dev-tunnels/_static/dev-tunnels-tool-window.png" alt-text="Dev Tunnels tool window.":::
+
+To open this window, Use the **Show Dev Tunnels Window** menu option in the debug dropdown. Or select **View** > **Other Windows** > **Dev Tunnels**.
+
+From this tool window, create a new tunnel by selecting the green `+` button. When you create a tunnel from the tool window, it isn't automatically set as the active tunnel like it is when a tunnel is created from the debug dropdown.
```

Comment:
```suggestion
From the **Dev Tunnels**  window, create a new tunnel by selecting the green `+` button. When a tunnel is created from the **Dev Tunnels** window, it isn't automatically set as the active tunnel like it is when a tunnel is created from the debug dropdown.
```
Then how to you make it the active tunnel?

Reply:
See next comment

Reply:
Currently you make it active by using the Debug dropdown, later you'll be able to do it in this tool window also.